include/LightGBM/c_api.h

```diff
@@ -389,6 +398,15 @@ LIGHTGBM_C_EXPORT int LGBM_DatasetGetNumData(DatasetHandle handle,
 LIGHTGBM_C_EXPORT int LGBM_DatasetGetNumFeature(DatasetHandle handle,
                                                 int* out);
 
+/*!
+ * \brief Get number of original features.
+ * \param handle Handle of dataset
+ * \param[out] out The address to hold number of features
+ * \return 0 when succeed, -1 when failure happens
+ */
+LIGHTGBM_C_EXPORT int LGBM_DatasetGetNumOriginalFeature(DatasetHandle handle,
```

Comment:
I remember that LGBM_DatasetGetNumFeature returns the number of original features. Maybe this additional api is not needed?

Reply:
cc @shiyu1994

Reply:
Because the CTR construction is before dataset construction, the LGBM_DatasetGetNumFeature returns the number of features with CTR transformations. So I think this is necessary.